### ISSUE TITLE
Incorrect ScopeIndent.IncorrectExact report for match inside array literal

### ISSUE THREAD

**Issue by TimWolla** [squizlabs/PHP_CodeSniffer#3672]
**Describe the bug**

phpcs in PSR12 mode reports that a line is indented incorrectly when it is correctly indented according to my understanding.

**Code sample**
```php
<?php

namespace foo;

class Foo
{
    public static function foo(array $foo): array
    {
        return \array_map(static function ($foo) {
            return [
                match ($foo) {
                    0 => 1,
                    1 => 0,
                },
                $foo,
            ];
        }, $foo);
    }
}
```

**Custom ruleset**

`--standard=PSR12`

**To reproduce**

1. Create a file called `test.php` with the code sample above...
2. Run `phpcs --standard=PSR12 -s test.php`
3. See error message displayed
```
FILE: /pwd/test.php
--------------------------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
--------------------------------------------------------------------------------------------------------------------------
 11 | ERROR | [x] Line indented incorrectly; expected 12 spaces, found 16
    |       |     (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
 14 | ERROR | [x] Line indented incorrectly; expected 12 spaces, found 16
    |       |     (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
--------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
--------------------------------------------------------------------------------------------------------------------------

Time: 73ms; Memory: 8MB
```

When reformatting with `phpcbf` the file will look like this afterwards:

```php
<?php

namespace foo;

class Foo
{
    public static function foo(array $foo): array
    {
        return \array_map(static function ($foo) {
            return [
            match ($foo) {
                0 => 1,
                1 => 0,
            },
                $foo,
            ];
        }, $foo);
    }
}
```

**Expected behavior**

I did not expect any errors in the given file.

I expected phpcbf not to change the file.

**Versions (please complete the following information):**
 - OS: Ubuntu 20.04, Docker.
 - PHP: 8.1
 - PHPCS: PHP_CodeSniffer version 3.7.1 (stable) by Squiz (http://www.squiz.net)
 - Standard: PSR12

**Additional context**

n/a

**Comment by gsherwood** [comment]
Looks to be something wrong going on here; possibly a conflict between ignoring exact indents and fixed indents for control structures.

Smallest sample code to replicate:
```php
return foo(function ($foo) {
    return [
        match ($foo) {
        }
    ];
});
```

**Comment by gsherwood** [comment]
I've pushed a fix for this issue, which will be in the 3.7.2 release. Thanks for the bug report.
